TESTS/network/wifi/wifi_connect_params_valid_secure.cpp

```diff
@@ -26,5 +26,22 @@ using namespace utest::v1;
 void wifi_connect_params_valid_secure(void)
 {
     WiFiInterface *wifi = get_interface();
-    TEST_ASSERT_EQUAL_INT(NSAPI_ERROR_OK, wifi->connect(MBED_CONF_APP_WIFI_SECURE_SSID, MBED_CONF_APP_WIFI_PASSWORD, NSAPI_SECURITY_WPA2));
+
+    if(wifi->connect(MBED_CONF_APP_WIFI_SECURE_SSID, MBED_CONF_APP_WIFI_PASSWORD, NSAPI_SECURITY_WPA2) == NSAPI_ERROR_OK) {
+        TEST_PASS();
+    }
+
+    if(wifi->connect(MBED_CONF_APP_WIFI_SECURE_SSID, MBED_CONF_APP_WIFI_PASSWORD, NSAPI_SECURITY_WPA_WPA2) == NSAPI_ERROR_OK) {
+        TEST_PASS();
+    }
+
+    if(wifi->connect(MBED_CONF_APP_WIFI_SECURE_SSID, MBED_CONF_APP_WIFI_PASSWORD, NSAPI_SECURITY_WPA) == NSAPI_ERROR_OK) {
+        TEST_PASS();
+    }
+
+    if(wifi->connect(MBED_CONF_APP_WIFI_SECURE_SSID, MBED_CONF_APP_WIFI_PASSWORD, NSAPI_SECURITY_WEP) == NSAPI_ERROR_OK) {
+        TEST_PASS();
+    }
```

Comment:
I don't think this works.

Our test environment does not have SSID that would allow you to connect with all these parameters.

You can only expect one to work.

Also, if first one works, how can we expect second `connect()` to work without disconnecting in between.

Reply:
Based on the fact that test case execution should abort immediately when TEST_PASS is called there shouldn't be an issue. WPA2 is tried first so this should behave the same in our environment as the previous solution.

    /* It is not necessary for you to call PASS. A PASS condition is assumed if nothing fails.
     * This method allows you to abort a test immediately with a PASS state, ignoring the remainder of the test. */
    #define TEST_PASS()    longjmp(Unity.AbortFrame, 1)

If connect() fails and is repeated with another security protocol this shouldn't be an issue as the failed try should only report error - which gets ignored. Only if none of the methods succeeds we have an error situation. Lets also remember that these test cases are not only to be run in our test environment.

Also a thing to keep in mind is that WIFI-SET-CREDENTIAL test case checks that all the expected security protocols are supported.

Reply:
> Our test environment does not have SSID that would allow you to connect with all these parameters.

Our test environment gets covered by the first if-clause, others are to prevent the need to modify the test case itself when someone else is using an access point with WEP/WPA/WPA_WPA2 Mixed mode.

Reply:
Actually, can you make if like:
```
if (...) {
    TEST_PASS();
    return;
}
```

just to make it more obvious.

Reply:
That's as simple as it can get.